source/state_representation/src/parameters/ParameterMap.cpp

```diff
@@ -52,9 +52,8 @@ void ParameterMap::assert_parameter_valid(const std::shared_ptr<ParameterInterfa
   }
 }
 
-void
-ParameterMap::validate_and_set_parameter(const std::shared_ptr<ParameterInterface>& parameter) {
-  this->parameters_[parameter->get_name()] = parameter;
+void ParameterMap::validate_and_set_parameter(const std::shared_ptr<ParameterInterface>& parameter) {
+  this->parameters_.insert_or_assign(parameter->get_name(), parameter);
```

Comment:
This means that if a parameter does not exist yet it will simply be silently added. Is that what we want?

Reply:
I see. In that case I guess this is all good.

Reply:
Ah I actually had this exact changed line in a local unpushed branch 😆 I had started a ParameterMap refactor to clearly differentiate adding, setting and validating parameters (inspired by Component parameters). But it would have quite a large effect on all derived classes and it's not a priority. For now Dominic summarised it correctly 👍